src/realm/object-store/class.hpp

```diff
@@ -0,0 +1,56 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_CLASS_HPP
+#define REALM_OS_CLASS_HPP
+
+#include <realm/obj.hpp>
+#include "realm/parser/keypath_mapping.hpp"
+#include "realm/parser/query_parser.hpp"
+#include <memory>
+
+namespace realm {
+
+class ObjectSchema;
+class Realm;
+class Query;
+
+class Class {
```

Comment:
I don't love the conflation of "Class" with "Table" in core. In my mind, the SDK is responsible for using DB tables (the logical, not physical concept) to produce magical classes. But that is an SDK concern. At this layer we are exposing tables (or namespaces, or collections, or some other name for a named group of rows).